Prose/ProseLib/Sources/App/AppReducer.swift

```diff
@@ -33,11 +33,17 @@ public let appReducer: Reducer<
     ).disabled(when: \.isMainWindowDisabled),
     Reducer { state, action, environment in
         func proceedToMainFlow(with credentials: Credentials) {
-            state.main = MainScreenState(jid: credentials.jid)
+            if state.isMainWindowRedacted {
+                state.main = MainScreenState(jid: credentials.jid)
+            } else {
+                state.main.sidebar.credentials = credentials.jid
+            }
+            state.isMainWindowRedacted = false
```

Comment:
This doesn't make sense anymore, but should I keep it?

Reply:
If it doesn't make sense, then no.